### PR TITLE
DEEP-76 publishes SMS for each microservice

### PR DESCRIPTION
**Pull request by ek-ok**

### CODE REVIEW

File: scraper_service/main.py [hunk 1]
```diff
@@ -1,79 +1,90 @@
+import json
+import logging
 from datetime import datetime
 from time import sleep
+
+from pytz import timezone
```
Comment (ojdajuiceman) [inline scraper_service/main.py]: Let's not use this for now, it's not native to AWS and therefore hard to import

File: scraper_service/main.py [hunk 2]
```diff
@@ -1,79 +1,90 @@
...
-        # write_to_dynamo(filename, phone, 0)
+        except ValueError as err:
+            logging.info(err)
+            fail_count += 1
+            sleep(1)
 
-    return outbound_message
+        # Publish failed message to downstream services
+        outbound_message['body'] = (
+                "Bad news:( Image was not successfully scraped from ShackCam."
```
Comment (ojdajuiceman) [inline scraper_service/main.py]: Make sure to add a space after the period, also before the ":("

File: scraper_service/main.py [hunk 3]
```diff
@@ -1,79 +1,90 @@
...
-    else:
-        # Scrape the image if shack cam is working
-        session = boto3.Session()
-        s3 = session.resource('s3')
-        bucket_name = 'deepshack'
-        bucket = s3.Bucket(bucket_name)
+            # Publish message to downstream services
+            publish_message(outbound_message, PREDICT_TOPIC_ARN)
+            outbound_message['body'] = (
+                "Good news! DeepShack is on your way."
```
Comment (ojdajuiceman) [inline scraper_service/main.py]: Add a space after the period.

File: scraper_service/main.py [hunk 4]
```diff
@@ -1,79 +1,90 @@
...
+                "Image was scraped from ShackCam and saved on AWS S3."
```
Comment (ojdajuiceman) [inline scraper_service/main.py]: Add a space after the period.

File: scraper_service/main.py [hunk 5]
```diff
@@ -1,79 +1,90 @@
...
-    filename = f'shackcam/{timestamp_str}' + phone + '.jpg'
+    # Receive message from upstream service
+    event_message = event['Records'][0]['Sns']['Message']
+    phone_number = event_message['phone_number'].split("B")[-1]
 
-    # Create message and send to predict_service_trigger
-    # on topic "triggerPredict"
-    outbound_message = {'image_id': filename,
-                        'phone': phone}
+    timestamp = datetime.now(timezone('EST'))
```
Comment (ojdajuiceman) [inline scraper_service/main.py]: Because we're not using pytz, replace this line with:

timestamp = datetime.now()

**Review by ojdajuiceman — APPROVED** [review]
Looks good, EK. I just recommended a few small changes (mainly pytz)

File: scraper_service/main.py (hunk 1 above)
Comment (ek-ok) [reply scraper_service/main.py]: All set

File: scraper_service/main.py (hunk 2 above)
Comment (ek-ok) [reply scraper_service/main.py]: All set

File: scraper_service/main.py (hunk 3 above)
Comment (ek-ok) [reply scraper_service/main.py]: All set

File: scraper_service/main.py (hunk 4 above)
Comment (ek-ok) [reply scraper_service/main.py]: All set

File: scraper_service/main.py (hunk 5 above)
Comment (ek-ok) [reply scraper_service/main.py]: All set

**Pull request merged**